hypha/settings/base.py

```diff
@@ -134,13 +134,9 @@
 TRANSITION_AFTER_REVIEWS = env.bool("TRANSITION_AFTER_REVIEWS", False)
 
 # Forces users to log in first in order to make an application.  This is particularly useful in conjunction
-# with ENABLE_REGISTRATION_WITHOUT_APPLICATION
+# with ENABLE_PUBLIC_SIGNUP
 FORCE_LOGIN_FOR_APPLICATION = env.bool("FORCE_LOGIN_FOR_APPLICATION", False)
```

Comment:
Should we rename and revert this setting?

Reply:
Yes. we can go it, and also make it backward compatible. But the enable_registration_without_application won't/doesn't make much sense.

Reply:
I meant the `FORCE_LOGIN_FOR_APPLICATION` setting. Switch it from default False to default True. I also think it can have a better name, especially with a new default.

Reply:
I like the "ENABLE_PUBLIC_SIGNUP" change.

Reply:
Ah. that totally make sense! We can go for it. Can't think of a better name, but going we can potentially just remove it.